syntax = "proto3";
package p2pinterface;

/*
 * Use following command to generate py stubs
 * Under pyquarkchain directory
 * protoc -Iquarkchain/proto --python_out=quarkchain quarkchain/proto/p2pinterface.proto 
 */

// Message payload exchanged between quarkchain core and p2p service.
message QuarkMessage {
    enum Type {
        UNKNOWN = 0;
        P2PHELLO = 1;
        OTHER = 2;
    }
    bool is_broadcast = 1;
    string peer_id = 2;
    Type type = 3;
    bytes payload = 4;
}
